File: pounders/py/general_h_funs/leastsquares.py
import numpy as np

def leastsquares(Cres,Gres,Hres):
    n, _, m = Hres.shape

    G = 2 * Gres @ Cres.T
    H = np.zeros((n,n))
    for i in range(m):
        H = H + Cres[i]*Hres[:,:,i]

    H = 2*H + 2*Gres @ Gres.T

    return G, H

File: pounders/m/general_h_funs/leastsquares.m
function [G, H] = leastsquares(Cres, Gres, Hres)
[n, ~, m] = size(Hres);

G = 2 * Gres * Cres';
H = zeros(n);
for i = 1:m
    H = H + Cres(i) * Hres(:, :, i);
end
H = 2 * H + 2 * (Gres * Gres');
